src/java.base/share/classes/com/sun/crypto/provider/KeyWrapCipher.java

```diff
@@ -296,6 +296,12 @@ private void implInit(int opmode, Key key, byte[] iv, SecureRandom random)
         if (keyBytes == null) {
             throw new InvalidKeyException("Null key");
         }
+
+        if (fixedKeySize != -1 && fixedKeySize != keyBytes.length) {
+            throw new InvalidKeyException("Invalid key length: " +
+                    keyBytes.length + " bytes, should be " + fixedKeySize);
```

Comment:
Is it necessary to check that the key algorithm is "AES", and the size is a valid AES key size, as the code in engineGetKeySize()?

Reply:
Makes sense to have a consistent check, I will refactor it. Thanks for the comment~